packages/bot-skeleton/src/services/tradeEngine/trade/Proposal.js

```diff
@@ -68,44 +68,44 @@ export default Engine =>
-                this.proposal_templates.map(proposal =>
-                    doUntilDone(() =>
-                        this.api.subscribeToPriceForContractProposal(proposal).catch(error => {
-                            // We intercept ContractBuyValidationError as user may have specified
-                            // e.g. a DIGITUNDER 0 or DIGITOVER 9, while one proposal may be invalid
-                            // the other is valid. We will error on Purchase rather than here.
-                            if (error?.name === 'ContractBuyValidationError') {
-                                this.data.proposals.push({
-                                    ...error.error.echo_req,
-                                    ...error.error.echo_req.passthrough,
-                                    error,
-                                });
-
-                                return null;
-                            }
-
-                            if (!has_informed_error) {
-                                has_informed_error = true;
-                                this.$scope.observer.emit('Error', error.error.error);
-                            }
+                this.proposal_templates.map(proposal => {
+                    doUntilDone(() => this.api.send(proposal)).catch(error => {
+                        // We intercept ContractBuyValidationError as user may have specified
+                        // e.g. a DIGITUNDER 0 or DIGITOVER 9, while one proposal may be invalid
+                        // the other is valid. We will error on Purchase rather than here.
+
+                        if (error.error.code === 'ContractBuyValidationError') {
+                            this.data.proposals.push({
+                                ...error.error.echo_req,
+                                ...error.echo_req.passthrough,
+                                error,
+                            });
+
                             return null;
-                        })
-                    )
-                )
+                        }
+                        if (!has_informed_error) {
+                            has_informed_error = true;
+                            this.$scope.observer.emit('Error', error);
+                        }
+                        return null;
+                    });
+                })
             );
         }
 
         observeProposals() {
-            this.listen('proposal', response => {
-                const { passthrough, proposal } = response;
-
-                if (
-                    this.data.proposals.findIndex(p => p.id === proposal.id) === -1 &&
-                    !this.data.forget_proposal_ids.includes(proposal.id)
-                ) {
-                    // Add proposals based on the ID returned by the API.
-                    this.data.proposals.push({ ...proposal, ...passthrough });
-                    this.checkProposalReady();
+            this.api.onMessage().subscribe(response => {
```

Comment:
Are we handling errors here?

Reply:
No, we only get observe proposal here and on requestProposals function we handle errors